include/atomics/sender_cadmium.hpp

```diff
@@ -0,0 +1,210 @@
+																typename definitions::ackIn> ;
+        using output_ports = std::tuple <typename definitions::packetSentOut, 
+																typename definitions::ackReceivedOut, typename definitions::dataOut> ;
+
+        // internal transition
+        void internal_transition()
+        {
+            if (state.ack)
+            {
+                if (state.packet_number < state.total_packet_number)
+                {
+                    state.packet_number++;
+                    state.ack = false;
+                    state.alt_bit = (state.alt_bit + 1) % 2;
+                    state.sending = true;
+                    state.model_active = true;
+                    state.next_internal = PREPARATION_TIME;
+                }
+                else
+                {
+                    state.model_active = false;
+                    state.next_internal = std::numeric_limits <TIME>::infinity();
+                }
+            }
+            else
+            {
+                if (state.sending)
+                {
+                    state.sending = false;
+                    state.model_active = true;
+                    state.next_internal = timeout;
+                }
+                else
+                {
+                    state.sending = true;
+                    state.model_active = true;
+                    state.next_internal = PREPARATION_TIME;
+                }
+            }
+        }
+
+        // external transition
+        void external_transition(TIME e,
+								 typename make_message_bags <input_ports>::type mbs)
+        {
+            if ((get_messages <typename definitions::controlIn> (mbs).size() + get_messages 
+																				<typename definitions::ackIn> (mbs).size()) > 1)
+				{
+					assert(false && "one message per time uniti");
+				}
+            for (const auto &x: get_messages <typename definitions::controlIn> (mbs))
+            {
+                if (state.model_active == false)
+                {
+                    state.total_packet_number = static_cast <int> (x.value);
+                    if (state.total_packet_number > 0)
+                    {
+                        state.packet_number = 1;
+                        state.ack = false;
+                        state.sending = true;
+                        state.alt_bit = state.packet_number % 2; //set initial alt_bit
+                        state.model_active = true;
+                        state.next_internal = PREPARATION_TIME;
+                    }
+                    else
+                    {
+                        if (state.next_internal != std::numeric_limits <TIME>::infinity())
+                        {
+                            state.next_internal = state.next_internal - e;
+                        }
+                    }
+                }
+            }
+            for (const auto &x: get_messages <typename definitions::ackIn> (mbs))
```

Comment:
All for statement  needs to follow coding conventions as specified by professor:
ie,
for  () {
   value = 2;
}
pls, apply it through out your code